graylog2-server/src/main/java/org/graylog2/streams/StreamRouterEngine.java

```diff
@@ -137,33 +143,63 @@ public String getFingerprint() {
      * @return the list of matching streams
      */
     public List<Stream> match(Message message) {
-        final Map<Stream, StreamMatch> matches = Maps.newHashMap();
-        final Set<Stream> timeouts = Sets.newHashSet();
-        final List<Stream> result = Lists.newArrayList();
-        final Set<String> fieldNames = message.getFieldNames();
-
-        // Execute the rules ordered by complexity. (fast rules first)
-        matchRules(message, presenceFields, presenceRules, matches);
-        // Only pass an intersection of the rules fields to avoid checking every field! (does not work for presence matching)
-        matchRules(message, Sets.intersection(fieldNames, exactFields), exactRules, matches);
-        matchRules(message, Sets.intersection(fieldNames, greaterFields), greaterRules, matches);
-        matchRules(message, Sets.intersection(fieldNames, smallerFields), smallerRules, matches);
-        // Execute regex rules with a timeout to prevent bad regexes to hang the processing.
-        matchRulesWithTimeout(message, Sets.intersection(fieldNames, regexFields), regexRules, matches, timeouts);
-
-        // Register failure for streams where rules ran into a timeout.
-        for (Stream stream : timeouts) {
-            streamFaultManager.registerFailure(stream);
-        }
+        final Set<Stream> result = Sets.newHashSet();
+        final Set<Stream> blackList = Sets.newHashSet();
```

Comment:
These sets could be initialized with the (maximum) number of matches: [`Sets#newHashSetWithExpectedSize(int)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Sets.html#newHashSetWithExpectedSize(int))